components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -0,0 +1,308 @@
+        &mut self,
+        ctx: &mut StoreContext<EK, ER, T>,
+        msg: eraftpb::Message,
+    ) -> Option<RaftMessage> {
+        let to_peer = match self.get_peer_from_cache(msg.to) {
+            Some(p) => p,
+            None => {
+                warn!(self.logger, "failed to look up recipient peer"; "to_peer" => msg.to);
+                return None;
+            }
+        };
+
+        let mut raft_msg = self.prepare_raft_message();
+
+        raft_msg.set_to_peer(to_peer);
+        if msg.from != self.peer().id {
+            debug!(
+                self.logger,
+                "redirecting message";
+                "msg_type" => ?msg.get_msg_type(),
+                "from" => msg.get_from(),
+                "to" => msg.get_to(),
+            );
+        }
+        raft_msg.set_message(msg);
+        Some(raft_msg)
+    }
+
+    /// Send a message.
+    ///
+    /// The message is pushed into the send buffer, it may not be sent out until
+    /// transport is flushed explicitly.
+    fn send_raft_message<T: Transport>(
+        &mut self,
+        ctx: &mut StoreContext<EK, ER, T>,
+        msg: RaftMessage,
+    ) {
+        let msg_type = msg.get_message().get_msg_type();
+        let to_peer_id = msg.get_to_peer().get_id();
+        let to_store_id = msg.get_to_peer().get_store_id();
+
+        trace!(
+            self.logger,
+            "send raft msg";
+            "msg_type" => ?msg_type,
+            "msg_size" => msg.get_message().compute_size(),
+            "to" => to_peer_id,
+        );
+
+        match ctx.trans.send(msg) {
+            Ok(()) => ctx.raft_metrics.send_message.add(msg_type, true),
+            Err(e) => {
+                // We use metrics to observe failure on production.
+                debug!(
+                    self.logger,
+                    "failed to send msg to other peer";
+                    "target_peer_id" => to_peer_id,
+                    "target_store_id" => to_store_id,
+                    "err" => ?e,
+                    "error_code" => %e.error_code(),
+                );
+                // unreachable store
+                self.raft_group_mut().report_unreachable(to_peer_id);
+                ctx.raft_metrics.send_message.add(msg_type, false);
+            }
+        }
+    }
+
+    fn handle_raft_committed_entries<T>(
+        &self,
+        _ctx: &mut crate::batch::StoreContext<EK, ER, T>,
+        _take_committed_entries: Vec<raft::prelude::Entry>,
+    ) {
+        unimplemented!()
+    }
+
+    /// Processing the ready of raft. A detail description of how it's handled
+    /// can be found at https://docs.rs/raft/latest/raft/#processing-the-ready-state.
+    ///
+    /// It's should be called at the end of every round of processing. Any
+    /// writes will be handled asynchronously, and be notified once writes
+    /// are persisted.
+    #[inline]
+    pub fn handle_raft_ready<T: Transport>(&mut self, ctx: &mut StoreContext<EK, ER, T>) {
```

Comment:
is this similar to handle_raft_ready_append in v1?

Reply:
Yes. There is no snapshot processing for now.